integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/events/DeleteSegmentByIdListener.scala

```diff
@@ -49,8 +51,13 @@ class DeleteSegmentByIdListener extends OperationEventListener with Logging {
           val table = metastore
             .lookupRelation(Some(carbonTable.getDatabaseName), tableName)(sparkSession)
             .asInstanceOf[CarbonRelation].carbonTable
-          CarbonStore
-            .deleteLoadById(loadIds, carbonTable.getDatabaseName, table.getTableName, table)
+          val dataLoadLocation = CarbonTablePath.getTableStatusFilePath(table.getTablePath)
+          // this check is added to check if the table status file exists or not. Delete on index
```

Comment:
rename `dataLoadLocation` to `tableStatusFilePath` , and please improve the comment, make it more meaningful to user, because its not index files, its index table, please update

Reply:
done